src/PerformanceTests/Utils/TaskHelper.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+public class TaskHelper
+{
+    static readonly int cores = Environment.ProcessorCount;
+
+    public static async Task ParallelFor(int count, Func<Task> action)
+    {
+        // If we have a batch of 10,000 and have 10 cores we do 10 iterations of 1,000 message chunks.
```

Comment:
Exactly what I said on the call. It was just a first hack. Not
sophisticated as you pointed out. The problem is that depending on the
latency and number of tasks it is better to chunk and sometimes not. I'm
fine if you guys say: let's not chunk

On Monday, April 18, 2016, Ramon Smits <notifications@github.com> wrote:

> In src/PerformanceTests/Utils/TaskHelper.cs
> <https://github.com/Particular/EndToEnd/pull/61#discussion_r60098039>:
>
> > @@ -0,0 +1,25 @@
> > +﻿using System;
> > +using System.Collections.Generic;
> > +using System.Threading.Tasks;
> > +
> > +public class TaskHelper
> > +{
> > +    static readonly int cores = Environment.ProcessorCount;
> > +
> > +    public static async Task ParallelFor(int count, Func<Task> action)
> > +    {
> > +        // If we have a batch of 10,000 and have 10 cores we do 10 iterations of 1,000 message chunks.
>
> @danielmarbach <https://github.com/danielmarbach> Was it really your
> intention to have this behavior? Super large sets would basically have the
> same behavior.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/EndToEnd/pull/61/files/abd2cd5b5e4122dd62555d09603fd491f3d16e91#r60098039>
>